sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -120,9 +120,9 @@ private async ValueTask ProcessAsync(HttpMessage message, bool async)
                 }
             }
             // HttpClient on NET5 throws OperationCanceledException from sync call sites, normalize to TaskCanceledException
-            catch (OperationCanceledException)
+            catch (OperationCanceledException e) when (CancellationTokenExtensions.ShouldWrapInOperationCanceledException(e, message.CancellationToken))
```

Comment:
You can't use the extension method as an extension method in this case?

Reply:
Wanted to copy CancellationHelper from BCL as close as possible but forgot to copy the name. With CancellationHelper name it should be less strange.